apps/api/src/app/widgets/usecases/update-subscriber-preference/update-subscriber-preference.usecase.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable } from '@nestjs/common';
+import { SubscriberPreferenceRepository, SubscriberPreferenceEntity } from '@novu/dal';
+import { UpdateSubscriberPreferenceCommand } from './update-subscriber-preference.command';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+
+@Injectable()
+export class UpdateSubscriberPreference {
+  constructor(private subscriberPreferenceRepository: SubscriberPreferenceRepository) {}
+
+  async execute(command: UpdateSubscriberPreferenceCommand): Promise<SubscriberPreferenceEntity> {
+    const userPreference = await this.subscriberPreferenceRepository.findOne({
+      _organizationId: command.organizationId,
+      _environmentId: command.environmentId,
+      _subscriberId: command.subscriberId,
+      _templateId: command.templateId,
+    });
+
+    if (!userPreference) {
+      const channelObj = {} as Record<'email' | 'sms' | 'in_app' | 'direct' | 'push', boolean>;
+      channelObj[command.channel?.type] = command.channel?.enabled;
+
+      return await this.subscriberPreferenceRepository.create({
+        _environmentId: command.environmentId,
+        _organizationId: command.organizationId,
+        _subscriberId: command.subscriberId,
+        _templateId: command.templateId,
+        enabled: command.enabled !== false,
+        channels: channelObj.hasOwnProperty('undefined') ? null : channelObj,
```

Comment:
```suggestion
        channels: !command.channel?.type ? null : channelObj,
```

the hasOwnProperty feel a bit strange, how is it possible that the user will send undefined channel type?

Reply:
You are right he can not, but at that moment of updating the subscriber preference we support the update of the following:
1. enabled (UpdateSubscriberPreferenceBodyDto)
2. channel (UpdateSubscriberPreferenceBodyDto)

due to the reason the user can choose what to update he can provide only the first one if he will provide only the 'enabled' that will be resulted with command.channel?.type of 'undefined'.

I agree It is a bit complex, what do you think?

Reply:
Oh I understand, but I would still assert based on the suggestion I made, I think it's easier to understand. Would worth creating a test case for this behaviour too I believe.

Reply:
Agree i liked your approach more. 
"should create user preference" tests if the user did not provide a channel.